### ISSUE TITLE
Question about Two Serials

### ISSUE THREAD

**Issue by samphonic** [akamali/mhk1_mqtt#3]
Hi there,

I'm building a [similar Mitsubishi interface](https://github.com/Sammy1Am/mitsubishi-uart) for ESPHome, with support for MHK(2) thermostat pass-through very much like what you've made, and had a quick question:

I'm noticing I get communication issues with the thermostat if there are too many simultaneous messages coming from the heat pump.  More specifically when my code refreshes the status of the heat pump, it sends and receives 4 different queries to the heat pump in quick succession (one packet sent, then one packet received, x4); and whenever this happens, I miss a message from the thermostat (and then it gets grumpy and has to go through its reconnect process again).

Did you ever see any issues with communications happening simultaneously on both serials?  It's not quite apples to apples as I'm using two software serials on an ESP8266, and you're using two hardware serials, but I figured I'd at least see if you came across the issue and if you did anything to attempt to mitigate it.

Thanks for the project and write-up, by the way, it was helpful seeing how you handled the interaction of MQTT and thermostat inputs.

**Comment by akamali** [comment]
I've had many serial issues, I've found things work best one message at a time. The protocol in general didn't seem like something that supports multiple messages in a quick succession, so I tried to avoid all that.

I also noticed the MHK1 has a very short time out window for its messages, short enough that it didn't leave enough time for me to send many messages in between messages sent by MHK1.

**Comment by samphonic** [comment]
Thanks for the pointers.  Your code comments about the MHK having a very short timeout window were super helpful because I was able to prioritize handling the MHK messages so it didn't drop off.

After posting here I did some digging through the SoftwareSerial code, and I think what ultimately was causing a lot of my issues might have been the interrupts for the two software serials stepping on each other at such a low baud rate.  I tried the same code on an ESP32 (using one of the hardware serials) and it seems much more stable.  I'll probably just stick to the ESP32 for units with a thermostat, and use my ESP8266s on the non-thermostat ones.

Thanks again.
